agent/client/client.go

```diff
@@ -109,6 +109,10 @@ func New(cfg *config.Config, supervisor supervisor, connectionChecker connection
 func (c *Client) Run(ctx context.Context) error {
 	c.l.Info("Starting...")
 
+	c.rw.Lock()
+	c.done = make(chan struct{})
+	c.rw.Unlock()
```

Comment:
We already created this channel in constructor

Reply:
we have to recreate on each reconnect

Reply:
If so then probably we can remove channel allocation from constructor? Or it's possible that this channel will be used before `Run` invocation?